pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -49,6 +49,24 @@ type InfinispanServiceContainerSpec struct {
 	StorageClassName string  `json:"storageClassName,optional,omitempty"`
 }
 
+// UpgradePolicy specifies all the possible policies that rule the ispn cluster upgrade
+type UpgradePolicy string
+
+const (
+	// UpgradePolicyManual only user can manually trigger a cluster upgrade
+	UpgradePolicyManual UpgradePolicy = "Manual"
+	// UpgradePolicyCVE cluster upgrade is triggered automatically only if a CVE upgrade is available
+	UpgradePolicyCVE UpgradePolicy = "CVE"
+	// UpgradePolicyLatest cluster upgrade is triggered automatically if a new image is available
+	UpgradePolicyLatest UpgradePolicy = "Latest"
+)
+
+// InfinispanUpgrade represents the upgrade strategy
+type InfinispanUpgrade struct {
+	// Policy represent the current upgrade policy for this cluster
+	Policy *UpgradePolicy `json:"strategy,optional,omitempty"`
```

Comment:
We don't need to have a pointer on simple types of we don't want to remove this field